### PR TITLE
Improve text chunk component

### PR DESCRIPTION
**Pull request by adrg**
- Add Wrap method for TextChunk
- Add Fit method for TextChunk
- Refactor the paragraph component to use the newly added Wrap method
- Add TextChunk test file

Addresses #178

---
This change is [Reviewable](https://reviewable.io/reviews/unidoc/unipdf/181)

### CODE REVIEW

File: creator/text_chunk.go [hunk 1]
```diff
@@ -26,17 +31,112 @@ type TextChunk struct {
...
+
+// Fit fits the chunk into the specified bounding box, cropping off the
+// remainder in a new chunk, if it exceeds the specified dimensions.
+func (tc *TextChunk) Fit(width, height float64) (*TextChunk, error) {
+	lines, err := tc.Wrap(width)
+	if err != nil {
+		return nil, err
+	}
+
+	fit := int(height / tc.Style.FontSize)
```
Comment (gunnsth) [inline creator/text_chunk.go]: Need to account for line heigth?  Cap height?

Comment (gunnsth) [reply creator/text_chunk.go]: Follow up: It was discussed that the line height indeed needs to be accounted for.  However the line height is not a property of the Chunk but rather the Paragraph that contains the Chunk.  The problem can be averted by dividing out the line height prior to calling Fit/Wrap methods (if line height is not 1.0 - default).
In the future we may consider making the LineHeight a property of Chunk and simply using the LineHeight of Paragraph elements as a wrapper to set the lineheight of Chunks.  Needs some investigation (a ticket will be created for this).

Comment (gunnsth) [reply creator/text_chunk.go]: Issue https://github.com/unidoc/unipdf/issues/183 tracks this

File: creator/text_chunk.go [hunk 2]
```diff
@@ -26,17 +31,112 @@ type TextChunk struct {
...
+	}
+	if len(line) > 0 {
+		lines = append(lines, string(line))
+	}
+
+	return lines, nil
+}
+
+// Fit fits the chunk into the specified bounding box, cropping off the
+// remainder in a new chunk, if it exceeds the specified dimensions.
```
Comment (gunnsth) [inline creator/text_chunk.go]: @adrg Can you add a note regarding the line height to the comment for Fit function.  Probably the only one that needs it?

Comment (adrg) [reply creator/text_chunk.go]: Sure, a note sounds like a good idea. Yes, the Wrap method does not need the line height.

**Pull request merged**